src/OpenTelemetry.Exporter.Geneva/README.md

```diff
@@ -1,6 +1,174 @@
 
-TBD
+The Geneva Exporter exports telemetry to
+[Event Tracing for Windows (ETW)](https://docs.microsoft.com/windows/win32/etw/about-event-tracing)
+or to a
+[Unix Domain Socket (UDS)](https://en.wikipedia.org/wiki/Unix_domain_socket)
+on the local machine.
+
+## Installation
+
+```shell
+dotnet add package OpenTelemetry.Exporter.Geneva
+```
+
+## Configuration
+
+The three types of telemetry are handled separately in OpenTelemetry.
+Therefore, each type of telemetry **must be** enabled separately.
+
+### Enable Logs
+
+Install the latest stable version of
+[`Microsoft.Extensions.Logging`](https://www.nuget.org/packages/Microsoft.Extensions.Logging/)
+
+```shell
+dotnet add package OpenTelemetry.Exporter.Geneva
+```
+
+This snippet shows how to configure the Geneva Exporter for logs
+
+```csharp
+using var loggerFactory = LoggerFactory.Create(loggingBuilder => loggingBuilder
+    .AddOpenTelemetry(openTelemetryLoggerOptions =>
+    {
+        openTelemetryLoggerOptions.AddGenevaLogExporter(genevaExporterOptions =>
+        {
+            genevaExporterOptions.ConnectionString = "EtwSession=OpenTelemetry";
+        });
+    }));
+```
+
+The above code must be in application startup. In case of ASP.NET Core
+applications, this should be in `ConfigureServices` of `Startup` class.
+For ASP.NET applications, this should be in `Global.aspx.cs`.
+
+Since OpenTelemetry .NET SDK is a
+[LoggingProvider](https://docs.microsoft.com/dotnet/core/extensions/logging-providers),
+use the built-in mechanism to apply [log
+filtering](https://docs.microsoft.com/dotnet/core/extensions/logging?tabs=command-line#how-filtering-rules-are-applied).
+This filtering lets you control the logs that are sent to each registered
+provider, including the OpenTelemetry provider. `OpenTelemetry` is the
+[alias](https://docs.microsoft.com/dotnet/api/microsoft.extensions.logging.provideraliasattribute)
+for `OpenTelemetryLoggerProvider`, that may be used when configuring filtering
+rules.
+
+**NOTE:** _Some application types (e.g. [ASP.NET
+Core](https://docs.microsoft.com/aspnet/core/fundamentals/logging/#configure-logging-1))
+have default logging settings. Please review them to make sure
+`OpenTelemetryLoggingProvider` is configured to receive logs of appropriate
+levels and category.
+
+### Enable Traces
+
+This snippet shows how to configure the Geneva Exporter for traces
+
+```csharp
+using var tracerProvider = Sdk.CreateTracerProviderBuilder()
+    .SetSampler(new AlwaysOnSampler())
+    .AddSource("DemoSource")
+    .AddGenevaTraceExporter(options => {
+        options.ConnectionString = "EtwSession=OpenTelemetry";
+    })
+    .Build();
+```
+
+The above code must be in application startup. In case of ASP.NET Core
+applications, this should be in `ConfigureServices` of `Startup` class.
+For ASP.NET applications, this should be in `Global.aspx.cs`.
+
+### GenevaExporterOptions (for logs and traces)
```

Comment:
Use uppercase for the first letter of Logs, Traces, and Metrics when using these words to denote the signal type.